components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -222,3 +222,26 @@ fun SessionManager.runWithSession(
     }
     return false
 }
+
+/**
+ * Tries to find a session with the provided session ID or uses the selected session and runs the block if found.
+ *
+ * @return True if the session was found and run successfully.
+ */
+fun SessionManager.runWithSessionIdOrSelected(
+    sessionId: String?,
+    block: SessionManager.(Session) -> Unit
+): Boolean {
+    var blockWasRun = false
+    sessionId?.let {
+        findSessionById(sessionId)?.let { session ->
+            block(session)
+            blockWasRun = true
+        }
+    }
+    selectedSession?.let {
+        block(it)
+        blockWasRun = true
+    }
+    return blockWasRun
```

Comment:
@Amejia481 Wait. Now that you are not returning early.. are you not running the block twice if there's a session id and a selected session?

Reply:
Fixing it!

Reply:
Thanks for the catch!

Reply:
https://github.com/mozilla-mobile/android-components/pull/2970